protocol/v2/qbft/instance/instance.go

```diff
@@ -162,7 +161,14 @@ func (i *Instance) BaseMsgValidation(msg *specqbft.SignedMessage) error {
 		return errors.Wrap(err, "invalid signed message")
 	}
 
-	if msg.Message.Round < i.State.Round {
+	// If a node gets a commit quorum before round change and other nodes get it after,
+	// then the node with quorum before wouldn't be able to receive next round messages from others
+	// due to slashing protection, unless we allow decided messages from previous round.
+	if decided := msg.Message.MsgType == specqbft.CommitMsgType && len(msg.Signers) > 1; decided {
+		if msg.Message.Round < i.State.Round-1 {
+			return errors.New("past round")
+		}
```

Comment:
I would think that theoretically decided messages should be able to be processed at any point

Reply:
I can make them be processed at any point. @MatheusFranco99 what's your opinion on this?

Reply:
done

Reply:
@MatheusFranco99 answered you?
I actually am curious about a second point of view

Reply:
Sry for the delay
From a security point of view/msg validation:
- it's fine if we are limiting the number of decided messages for a round a peer can send. And, if I'm not mistaken, we are
From a protocol point of view:
- it's possible it the receiving node is extremely early